big_tests/tests/mod_http_notification_SUITE.erl

```diff
@@ -151,44 +125,94 @@ simple_message_no_listener(Config) ->
-    escalus:story(Config, [{alice, 1}],
-        fun(Alice) -> escalus:send(Alice, escalus_stanza:chat_to(BobJid, Msg)) end),
-
+    {ok, Alice} = escalus_client:start(Config, alice, <<"res1">>),
+    escalus:send(Alice, escalus_stanza:presence(<<"available">>)),
+    BobJid = escalus_users:get_jid(Config, bob),
+    Stanza = escalus_stanza:chat_to(BobJid, Msg),
+    escalus:send(Alice, Stanza),
+    escalus_client:stop(Config, Alice),
     %% Bob logs in
-    Bob = login_send_presence(Config, bob),
-
+    {ok, Bob} = escalus_client:start(Config, bob, <<"res1">>),
+    escalus:send(Bob, escalus_stanza:presence(<<"available">>)),
     %% He receives his initial presence and the message
     Stanzas = escalus:wait_for_stanzas(Bob, 2),
-%%  ct:pal("Stanzas:~p", [Stanzas]),
     escalus_new_assert:mix_match([is_presence, is_chat(Msg)], Stanzas),
     escalus_client:stop(Config, Bob).
 
-
-%%%===================================================================
-%%% Custom predicates
-%%%===================================================================
-
-is_chat(Content) ->
-    fun(Stanza) -> escalus_pred:is_chat_message(Content, Stanza) end.
-
 %%%===================================================================
 %%% Helpers
 %%%===================================================================
 
+get_http_request() ->
+    Key = got_http_request,
+    mongoose_helper:wait_until(
+      fun() -> 1 =:= length(ets:lookup(?ETS_TABLE, Key)) end,
+      true, #{name => missing_request}),
+    [Bins] = lists:map(fun({_, El}) -> El end, ets:lookup(?ETS_TABLE, Key)),
+    ets:delete(?ETS_TABLE, Key),
+    Bins.
+
 login_send_presence(Config, User) ->
     Spec = escalus_users:get_userspec(Config, User),
     {ok, Client} = escalus_client:start(Config, Spec, <<"dummy">>),
     escalus:send(Client, escalus_stanza:presence(<<"available">>)),
     Client.
 
-get_prefix(mod_http_notification_tests) ->
+is_chat(Content) ->
+    fun(Stanza) -> escalus_pred:is_chat_message(Content, Stanza) end.
+
+get_prefix(no_prefix) ->
     "/";
-get_prefix(mod_http_notification_tests_with_prefix) ->
+get_prefix(with_prefix) ->
     "/prefix";
 get_prefix(Config) ->
     GroupName = proplists:get_value(name, proplists:get_value(tc_group_properties, Config)),
     get_prefix(GroupName).
+
+start_pool() ->
+    HTTPOpts = [{server, http_notifications_host()}],
+    PoolOpts = [{strategy, available_worker}, {workers, 1}],
+    ejabberd_node_utils:call_fun(mongoose_wpool, start_configured_pools,
+                                 [[{http, global, http_pool, PoolOpts, HTTPOpts}]]).
+
+stop_pool() ->
+    ejabberd_node_utils:call_fun(mongoose_wpool, stop, [http, global, http_pool]).
+
+set_modules(Config0, Opts) ->
+    Config = dynamic_modules:save_modules(host(), Config0),
+    ModOpts = [{worker_timeout, 500}, {host, http_notifications_host()}] ++ Opts,
+    dynamic_modules:ensure_modules(host(), [{mod_http_notification, ModOpts}]),
```

Comment:
At the moment, the opts given to `mod_http_notification` are directly passed to `mod_event_pusher` in https://github.com/esl/MongooseIM/blob/f82a1aae6b9fb0c030e87fd30b6b8e39d6d07c0e/src/mod_http_notification.erl#L10-L11, as far as I understand this will still work the same way 🤔

Reply:
Yes, and I may be mistaken, but I don't see a place where these options would be read. For example, in the backend:
https://github.com/esl/MongooseIM/blob/f82a1aae6b9fb0c030e87fd30b6b8e39d6d07c0e/src/event_pusher/mod_event_pusher_http.erl#L52
This made me think that these options are discarded. But it is not really in the scope of this PR and I'm really not sure, just an observation.

Reply:
Yeah, this one is very tricky. Opts are stored globally on an ets table on mod_event_pusher: https://github.com/esl/MongooseIM/blob/f82a1aae6b9fb0c030e87fd30b6b8e39d6d07c0e/src/event_pusher/mod_event_pusher.erl#L62-L65

Reply:
And then it gets weirder and weirder, but yeah, that's out of the scope of this PR maybe 😅